src/main/java/edu/tseidler/states/OXGame.java

```diff
@@ -30,4 +51,117 @@ public void start() {
+                defaultProperties.get("player2_name"));
+        BoardField player2Mark = player1Mark.other();
+        boolean player2First = !player1First;
+        playerList.add(new Player(player1Name, player1Mark, player1First));
+        playerList.add(new Player(player2Name, player2Mark, player2First));
+        return playerList;
+    }
+
+    static Language loadLanguage(String langShort) {
+        if (!available.contains(langShort))
+            if (!available.contains(properties.get("language")))
+                langShort = defaultProperties.get("language");
+            else
+                langShort = (String) properties.get("language");
+        return new Language(langShort);
+    }
+
+    static Set<String> available = new HashSet<String>() {{
+        add("en");
+        add("pl");
+    }};
+
+    private static Map<String, String> defaultProperties = new HashMap<String, String>() {{
+        put("input", "stdin");
+        put("output", "stdout");
+        put("language", "en");
+        put("player1_name", "Jacek");
+        put("player1_mark", "X");
+        put("player1_first", "true");
+        put("player2_name", "Placek");
+        put("board_rows", "3");
+        put("board_cols", "3");
+        put("board_winn", "3");
+    }};
+
+    private Board getBoard() {
+        int board_rows = Integer.valueOf(defaultProperties.get("board_rows"));
+        int board_cols = Integer.valueOf(defaultProperties.get("board_cols"));
+        int board_winn = Integer.valueOf(defaultProperties.get("board_winn"));
+        try {
+            board_rows = Integer.valueOf((String) properties.get("board_rows"));
+            board_cols = Integer.valueOf((String) properties.get("board_cols"));
+            board_winn = Integer.valueOf((String) properties.get("board_winn"));
+        } catch (Exception e) {
+            logger.log(Level.ERROR, "error parsing board parameters from settings file");
+        }
+        BoardParameters parameters = new BoardParameters(board_rows, board_cols, board_winn);
+        return new Board(parameters);
+    }
+
+    private Consumer<String> getStringConsumer() {
+        Map<String, Consumer<String>> consumerMap = new HashMap<String, Consumer<String>>() {{
+            put("stdout", DEFAULT_STRING_CONSUMER);
+            put("stderr", ERROR_CONSUMER);
+            try {
+                Path path = Paths.get((String) properties.getOrDefault("output_file", "output.log"));
+                put("file", new FileLineConsumer(path));
+            } catch (IOException e) {
+                logger.log(Level.WARN, "file operation error - falling back to console output");
+                put("file", DEFAULT_STRING_CONSUMER);
+            }
+        }};
+
+        return consumerMap.get(properties.get("output"));
+    }
+
+    private Supplier<String> getStringSupplier() {
+        Map<String, Supplier<String>> supplierMap = new HashMap<String, Supplier<String>>() {{
```

Comment:
self comment:
extract static map from method

Reply:
to jednak musi tak zostać, ponieważ inaczej nie wczytam ustawień z pliku wcześniej :)